### PR TITLE
[mxnet][eia][build]Add MXNet 1.7.0 for Elastic Inference

### PR DESCRIPTION
**Pull request by ycnie**
*Issue #, if available:*

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet] | [ei] | [build]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [x] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [x] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
Added MXNet 1.5.1 Dockerfile for Elastic Inference, and related test modifications to allow the use of `eimx` accelerator library.
Default framework version on buildspec-eia.yml is now 1.7.0.

*Tests run:*
GitHub Repo PR tests.
*DLC image/dockerfile:*
MXNet 1.7.0 Elastic Inference Dockerfile: `Dockerfile.eia`
*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/sagemaker_tests/mxnet/inference/integration/sagemaker/test_elastic_inference.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 DEFAULT_HANDLER_PATH = os.path.join(RESOURCE_PATH, 'default_handlers')
 MODEL_PATH = os.path.join(DEFAULT_HANDLER_PATH, 'model.tar.gz')
-SCRIPT_PATH = os.path.join(DEFAULT_HANDLER_PATH, 'model', 'code', 'empty_module.py')
+SCRIPT_PATH = os.path.join(DEFAULT_HANDLER_PATH, 'model', 'code', 'eia_module.py')
```
Comment (tejaschumbalkar) [inline test/sagemaker_tests/mxnet/inference/integration/sagemaker/test_elastic_inference.py:26]: Is this compatible for EIA v1.5.1 image test?

File: test/test_utils/__init__.py [hunk 2]
```diff
@@ -321,7 +321,8 @@ def get_inference_run_command(image_uri, model_names, processor="cpu"):
     server_type = get_inference_server_type(image_uri)
     if processor == "eia":
         multi_model_location = {
-            "resnet-152-eia": "https://s3.amazonaws.com/model-server/model_archive_1.0/resnet-152-eia.mar",
+            "resnet-152-eia": "https://s3.amazonaws.com/model-server/model_archive_1.0/resnet-152-eia-1-7-0.mar",
+            "resnet-152-eia-1-5-1": "https://s3.amazonaws.com/model-server/model_archive_1.0/resnet-152-eia-1-5-1.mar",
```
Comment (tejaschumbalkar) [inline test/test_utils/__init__.py:325]: Add logic to use model name as per the image version (v1.5.1 and others) for ECS [test](https://github.com/aws/deep-learning-containers/blob/master/test/dlc_tests/ecs/mxnet/inference/test_ecs_mxnet_inference.py#L38) and EC2 [test](https://github.com/aws/deep-learning-containers/blob/master/test/dlc_tests/ec2/mxnet/inference/test_mxnet_inference.py#L12)

**Comment by tejaschumbalkar** [comment]
Can you also update the description on the PR

File: mxnet/inference/docker/1.7.0/py3/Dockerfile.eia [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+ && tar -xvf /opt/ei_tools_${HEALTH_CHECK_VERSION}.tar.gz -C /opt/ \
+ && rm -rf /opt/ei_tools_${HEALTH_CHECK_VERSION}.tar.gz \
+ && chmod a+x /opt/ei_tools/bin/health_check
+
+RUN pip install --no-cache-dir --upgrade \
+    pip \
+    setuptools
+
+RUN pip install --no-cache-dir \
+    mxnet==1.7.0.post1 \
```
[the remark this answers is not in the document]
Comment (ycnie) [reply mxnet/inference/docker/1.7.0/py3/Dockerfile.eia]: upgraded in Docker file

File: mxnet/inference/docker/1.7.0/py3/Dockerfile.eia [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    https://amazonei-apachemxnet.s3.amazonaws.com/eimx-1.0-py2.py3-none-manylinux1_x86_64.whl \
+    multi-model-server==$MMS_VERSION \
+    keras-mxnet==2.2.4.1 \
```
[the remark this answers is not in the document]
Comment (ycnie) [reply mxnet/inference/docker/1.7.0/py3/Dockerfile.eia]: tried to upgrade to the latest version for `keras-mxnet` and `onnx`. kept numpy the same

File: test/sagemaker_tests/mxnet/inference/integration/sagemaker/test_elastic_inference.py (hunk 1 above)
Comment (ycnie) [reply test/sagemaker_tests/mxnet/inference/integration/sagemaker/test_elastic_inference.py]: Added the logic to use different `entry_point` for `1.7.0` and `1.5.1`

File: test/test_utils/__init__.py (hunk 2 above)
Comment (ycnie) [reply test/test_utils/__init__.py:325]: good point. Added the logic to decide `model_name` by framework version

File: test/dlc_tests/ec2/mxnet/inference/test_mxnet_inference.py [hunk 5]
```diff
@@ -2,14 +2,14 @@
...
-from test.test_utils import CONTAINER_TESTS_PREFIX
+from test.test_utils import CONTAINER_TESTS_PREFIX, get_framework_and_version_from_tag
 from test.test_utils.ec2 import get_ec2_instance_type, execute_ec2_inference_test, get_ec2_accelerator_type
 from test.dlc_tests.conftest import LOGGER
 
 
 SQUEEZENET_MODEL = "squeezenet"
 BERT_MODEL = "bert_sst"
-RESNET_EIA_MODEL = "resnet-152-eia"
+DEFAULT_RESNET_EIA_MODEL = "resnet-152-eia"
```
Comment (tejaschumbalkar) [inline test/dlc_tests/ec2/mxnet/inference/test_mxnet_inference.py:12]: The model name is also used in [L68](https://github.com/tejaschumbalkar/deep-learning-containers/blob/master/test/dlc_tests/ec2/mxnet/inference/test_mxnet_inference.py#L68) for `get_inference_run_command` function. 

Can we resolve the model name in the pytest function itself and pass it to the `run_ec2_mxnet_inference` function or some other way?

Comment (ycnie) [reply test/dlc_tests/ec2/mxnet/inference/test_mxnet_inference.py]: sounds like a plan. Will resolve the model name inside the pytest functions

File: src/config/build_config.py [hunk 6]
```diff
@@ -1,6 +1,6 @@
 # Please only set it to True if you are preparing an EI related PR
 # Do remember to revert it back to False before merging any PR (including EI dedicated PR)
-ENABLE_EI_MODE = False
+ENABLE_EI_MODE = True
```
Comment (tejaschumbalkar) [inline src/config/build_config.py:3]: Lets revert this config.

**Review by tejaschumbalkar — COMMENTED** [review]
LGTM. Will approve once the build configs are reverted.

**Review by tejaschumbalkar — APPROVED** [review]

**Pull request merged**